init.lua

```diff
@@ -111,9 +111,68 @@ end
 
 
 --- Receiving ---
+-- IRCv3.2 tags
+local escapers = {
+	["s"] = " ";
+	["r"] = "\r";
+	["n"] = "\n";
+	[";"] = ";";
+	["\\"] = "\\";
+}
+local function parse_tags(tag_message)
+	local tags = {}
+	local cur_name
+	local charbuf = {}
+	local pos = 1
+	message_len = #tag_message
+	while pos <= message_len do
+		if tag_message:match("^\\", pos) then
+			local lookahead = tag_message:sub(pos+1, pos+1)
+			charbuf[#charbuf+1] = escapers[lookahead] or lookahead
+			pos = pos + 2
+		elseif cur_name then
+			if tag_message:match("^;", pos) then
+				tags[cur_name] = table.concat(charbuf)
+				cur_name = nil
+				charbuf = {}
+				pos = pos + 1
+			else
+				charbuf[#charbuf+1], pos = tag_message:match("([^\\;]+)()", pos)
+			end
+		else
+			if tag_message:match("^=", pos) then
+				if #charbuf > 0 then
+					cur_name = table.concat(charbuf)
+					charbuf = {}
+				end
+				pos = pos + 1
+			elseif tag_message:match("^;", pos) then
+				if #charbuf > 0 then
+					tags[table.concat(charbuf)] = true
+					charbuf = {}
+				end
+				pos = pos + 1
+			else
+				charbuf[#charbuf+1], pos = tag_message:match("([^\\=;]+)()", pos)
+			end
+		end
+	end
+	return tags
+end
+
 -- Main --
 -- http://calebdelnay.com/blog/2010/11/parsing-the-irc-message-format-as-a-client
-local function parse_message(message)
+local function parse_message(message_tagged)
+	-- Tags
+	local tagged, tags, message
```

Comment:
Is the variable `tagged` assigned to or used anywhere? I don't see anywhere it's actually used...